File: react-native/src/ffmpeg-kit-extended.ts
import NativeFFmpegKitExtended from './NativeFFmpegKitExtended';
import {
  FFmpegSession,
  FFplaySession,
  FFprobeSession,
  MediaInformationSession,
  parseSessionJson,
  parseSessionsJson,
  type Session,
} from './session';
import {SessionQueueManager} from './session-queue-manager';
import {LogLevel, Signal} from './types';

/**
 * Central lifecycle, session-history, build-introspection, and native utility
 * facade.
 *
 * Call `initialize()` exactly once near application startup before invoking
 * FFmpegKit APIs. Initialization is idempotent in JavaScript. Most command
 * execution should go through `FFmpegKit`, `FFprobeKit`, or `FFplayKit`; the
 * session factory methods here are useful for framework-style integrations.
 */
export class FFmpegKitExtended {
  private static initializedValue = false;

  /**
   * Loads and initializes the configured native FFmpegKit bundle.
   * Repeated calls are ignored after successful JavaScript initialization.
   */
  static initialize(): void {
    if (this.initializedValue) return;
    NativeFFmpegKitExtended.initialize();
    this.initializedValue = true;
  }

  /** Whether `initialize()` has completed in this JavaScript runtime. */
  static get initialized(): boolean {
    return this.initializedValue;
  }

  /** Throws a descriptive error when initialization has not occurred. */
  static requireInitialized(): void {
    if (!this.initializedValue) {
      throw new Error(
        'FFmpegKitExtended has not been initialized. Call FFmpegKitExtended.initialize() before using the package.',
      );
    }
  }

  /** Returns the native bundle build stamp used for diagnostics. */
  static getBuildStamp(): string {
    this.requireInitialized();
    return NativeFFmpegKitExtended.getBuildStamp();
  }

  /** Creates an unstarted FFmpeg processing session. */
  static createFFmpegSession(command: string): FFmpegSession {
    this.requireInitialized();
    return new FFmpegSession(
      NativeFFmpegKitExtended.createFFmpegSession(command),
      command,
    );
  }

  /** Creates an unstarted FFprobe command session. */
  static createFFprobeSession(command: string): FFprobeSession {
    this.requireInitialized();
    return new FFprobeSession(
      NativeFFmpegKitExtended.createFFprobeSession(command),
      command,
    );
  }

  /** Creates an unstarted FFplay session with a native timeout in milliseconds. */
  static createFFplaySession(command: string, timeoutMs = 500): FFplaySession {
    this.requireInitialized();
    return new FFplaySession(
      NativeFFmpegKitExtended.createFFplaySession(command),
      command,
      timeoutMs,
    );
  }

  /**
   * Creates an unstarted structured media-information session from a complete
   * FFprobe command string.
   */
  static createMediaInformationSession(
    command: string,
    timeoutMs = 500,
  ): MediaInformationSession {
    this.requireInitialized();
    return new MediaInformationSession(
      NativeFFmpegKitExtended.createMediaInformationSession(command),
      command,
      timeoutMs,
    );
  }

  /** Requests cancellation by native session ID. */
  static cancelSession(sessionId: number): void {
    this.requireInitialized();
    NativeFFmpegKitExtended.cancelSession(sessionId);
  }

  /** Clears queued work and requests cancellation of all active sessions. */
  static cancelAllSessions(): void {
    this.requireInitialized();
    SessionQueueManager.shared.cancelAll();
  }

  /** Returns all sessions retained in native history. */
  static getSessions(): Session[] {
    this.requireInitialized();
    return parseSessionsJson(NativeFFmpegKitExtended.getSessionsJson('all'));
  }

  /** Alias of `getSessions()`. */
  static listSessions(): Session[] {
    return this.getSessions();
  }

  /** Returns retained FFmpeg processing sessions. */
  static getFFmpegSessions(): FFmpegSession[] {
    return parseSessionsJson(
      NativeFFmpegKitExtended.getSessionsJson('ffmpeg'),
    ).filter((session): session is FFmpegSession => session.isFFmpegSession());
  }

  /** Returns retained FFprobe command sessions. */
  static getFFprobeSessions(): FFprobeSession[] {
    return parseSessionsJson(
      NativeFFmpegKitExtended.getSessionsJson('ffprobe'),
    ).filter((session): session is FFprobeSession => session.isFFprobeSession());
  }

  /** Returns retained FFplay sessions. */
  static getFFplaySessions(): FFplaySession[] {
    return parseSessionsJson(
      NativeFFmpegKitExtended.getSessionsJson('ffplay'),
    ).filter((session): session is FFplaySession => session.isFFplaySession());
  }

  /** Returns retained structured media-information sessions. */
  static getMediaInformationSessions(): MediaInformationSession[] {
    return parseSessionsJson(
      NativeFFmpegKitExtended.getSessionsJson('media-information'),
    ).filter(
      (session): session is MediaInformationSession =>
        session.isMediaInformationSession(),
    );
  }

  /** Looks up one retained session by ID. */
  static getSession(sessionId: number): Session | undefined {
    this.requireInitialized();
    return parseSessionJson(NativeFFmpegKitExtended.getSessionJson(sessionId));
  }

  /** Returns the newest retained session of any type. */
  static getLastSession(): Session | undefined {
    this.requireInitialized();
    return parseSessionJson(NativeFFmpegKitExtended.getLastSessionJson('all'));
  }

  /** Returns the newest retained FFmpeg session. */
  static getLastFFmpegSession(): FFmpegSession | undefined {
    const session = parseSessionJson(
      NativeFFmpegKitExtended.getLastSessionJson('ffmpeg'),
    );
    return session?.isFFmpegSession() ? session : undefined;
  }

  /** Returns the newest retained FFprobe session. */
  static getLastFFprobeSession(): FFprobeSession | undefined {
    const session = parseSessionJson(
      NativeFFmpegKitExtended.getLastSessionJson('ffprobe'),
    );
    return session?.isFFprobeSession() ? session : undefined;
  }

  /** Returns the newest retained FFplay session. */
  static getLastFFplaySession(): FFplaySession | undefined {
    const session = parseSessionJson(
      NativeFFmpegKitExtended.getLastSessionJson('ffplay'),
    );
    return session?.isFFplaySession() ? session : undefined;
  }

  /** Returns the newest retained media-information session. */
  static getLastMediaInformationSession(): MediaInformationSession | undefined {
    const session = parseSessionJson(
      NativeFFmpegKitExtended.getLastSessionJson('media-information'),
    );
    return session?.isMediaInformationSession() ? session : undefined;
  }

  /** Enables native log redirection used by per-session callbacks. */
  static enableRedirection(): void {
    NativeFFmpegKitExtended.enableRedirection();
  }

  /** Disables native log redirection. */
  static disableRedirection(): void {
    NativeFFmpegKitExtended.disableRedirection();
  }

  /** Sets the process-wide FFmpeg log level. */
  static setLogLevel(level: LogLevel): void {
    NativeFFmpegKitExtended.setLogLevel(level);
  }

  /** Returns the process-wide FFmpeg log level. */
  static getLogLevel(): LogLevel {
    return NativeFFmpegKitExtended.getLogLevel() as LogLevel;
  }

  /** Registers fonts for FFmpeg filters, with an optional family-to-file map. */
  static setFontDirectory(path: string, mapping?: Record<string, string>): void {
    NativeFFmpegKitExtended.setFontDirectory(
      path,
      mapping ? JSON.stringify(mapping) : '',
    );
  }

  /** Selects a native audio output device by backend-provided name. */
  static setAudioOutputDevice(deviceName: string): void {
    NativeFFmpegKitExtended.setAudioOutputDevice(deviceName);
  }

  /** Lists native audio output devices in backend-defined text form. */
  static listAudioOutputDevices(): string {
    return NativeFFmpegKitExtended.listAudioOutputDevices();
  }

  /** Sets a native environment variable for FFmpeg and linked libraries. */
  static setEnvironmentVariable(name: string, value: string): void {
    NativeFFmpegKitExtended.setEnvironmentVariable(name, value);
  }

  /** Configures the native runtime to ignore a supported signal. */
  static ignoreSignal(signal: Signal): void {
    NativeFFmpegKitExtended.ignoreSignal(signal);
  }

  /** Returns the bundled upstream FFmpeg version. */
  static getFFmpegVersion(): string {
    return NativeFFmpegKitExtended.getFFmpegVersion();
  }

  /** Returns the architecture reported by the native FFmpeg build. */
  static getFFmpegArchitecture(): string {
    return NativeFFmpegKitExtended.getFFmpegArchitecture();
  }

  /** Returns the FFmpegKit Extended wrapper version. */
  static getVersion(): string {
    return NativeFFmpegKitExtended.getVersion();
  }

  /** Returns the configured native package name. */
  static getPackageName(): string {
    return NativeFFmpegKitExtended.getPackageName();
  }

  /** Lists external libraries enabled in the selected native bundle. */
  static getExternalLibraries(): string {
    return NativeFFmpegKitExtended.getExternalLibraries();
  }

  /** Returns the selected bundle type, such as shared/static build metadata. */
  static getBundleType(): string {
    return NativeFFmpegKitExtended.getBundleType();
  }

  /** Whether the selected native bundle includes GPL components. */
  static isGpl(): boolean {
    return NativeFFmpegKitExtended.isGpl();
  }

  /** Whether the selected native bundle includes nonfree components. */
  static isNonfree(): boolean {
    return NativeFFmpegKitExtended.isNonfree();
  }

  /** Returns the native FFmpeg codec registry text. */
  static getRegisteredCodecs(): string {
    return NativeFFmpegKitExtended.getRegisteredCodecs();
  }

  /** Returns the native FFmpeg encoder registry text. */
  static getRegisteredEncoders(): string {
    return NativeFFmpegKitExtended.getRegisteredEncoders();
  }

  /** Returns the native FFmpeg decoder registry text. */
  static getRegisteredDecoders(): string {
    return NativeFFmpegKitExtended.getRegisteredDecoders();
  }

  /** Returns the native FFmpeg muxer registry text. */
  static getRegisteredMuxers(): string {
    return NativeFFmpegKitExtended.getRegisteredMuxers();
  }

  /** Returns the native FFmpeg demuxer registry text. */
  static getRegisteredDemuxers(): string {
    return NativeFFmpegKitExtended.getRegisteredDemuxers();
  }

  /** Returns the native FFmpeg filter registry text. */
  static getRegisteredFilters(): string {
    return NativeFFmpegKitExtended.getRegisteredFilters();
  }

  /** Returns the native FFmpeg protocol registry text. */
  static getRegisteredProtocols(): string {
    return NativeFFmpegKitExtended.getRegisteredProtocols();
  }

  /** Returns the native FFmpeg bitstream-filter registry text. */
  static getRegisteredBitstreamFilters(): string {
    return NativeFFmpegKitExtended.getRegisteredBitstreamFilters();
  }

  /** Returns the upstream FFmpeg configure command/options. */
  static getBuildConfiguration(): string {
    return NativeFFmpegKitExtended.getBuildConfiguration();
  }

  /** Returns the native bundle build date text. */
  static getBuildDate(): string {
    return NativeFFmpegKitExtended.getBuildDate();
  }

  /** Sets the number of sessions retained in native history. */
  static setSessionHistorySize(size: number): void {
    NativeFFmpegKitExtended.setSessionHistorySize(size);
  }

  /** Returns the native session-history capacity. */
  static getSessionHistorySize(): number {
    return NativeFFmpegKitExtended.getSessionHistorySize();
  }

  /** Clears retained native session history. */
  static clearSessions(): void {
    NativeFFmpegKitExtended.clearSessions();
  }

  /** Creates a native FFmpeg FIFO/pipe and returns its path when supported. */
  static registerNewFFmpegPipe(): string | undefined {
    return NativeFFmpegKitExtended.registerNewFFmpegPipe() || undefined;
  }

  /** Closes/removes a pipe created by `registerNewFFmpegPipe()`. */
  static closeFFmpegPipe(path: string): void {
    NativeFFmpegKitExtended.closeFFmpegPipe(path);
  }

  /** Returns native callback messages still in transit for a session. */
  static messagesInTransmit(sessionId: number): number {
    return NativeFFmpegKitExtended.messagesInTransmit(sessionId);
  }
}

File: react-native/windows/FFmpegKitExtended/FFmpegKitExtended.cpp
#include "pch.h"
#include "FFmpegKitExtended.h"

#include "FFmpegKitDynamicApi.h"

#include <exception>
#include <string>
#include <utility>

namespace winrt::FFmpegKitExtended {
namespace {

[[noreturn]] void failFast(const char *method, const char *message) noexcept {
  std::string text = "FFmpegKitExtended Windows native call failed in ";
  text += method;
  text += ": ";
  text += message ? message : "unknown native error";
  text += "\n";
  OutputDebugStringA(text.c_str());
  RaiseFailFastException(nullptr, nullptr, 0);
  std::terminate();
}

template <typename Fn>
void invokeVoid(const char *method, Fn &&fn) noexcept {
  try {
    std::forward<Fn>(fn)();
  } catch (const std::exception &error) {
    failFast(method, error.what());
  } catch (...) {
    failFast(method, "non-standard exception");
  }
}

template <typename Result, typename Fn>
Result invoke(const char *method, Fn &&fn) noexcept {
  try {
    return std::forward<Fn>(fn)();
  } catch (const std::exception &error) {
    failFast(method, error.what());
  } catch (...) {
    failFast(method, "non-standard exception");
  }
}

} // namespace

namespace api = ffmpegkit::bridge;

void FFmpegKitExtended::initialize() noexcept {
  invokeVoid("initialize", [&] { api::initialize(); });
}

std::string FFmpegKitExtended::getBuildStamp() noexcept {
  return invoke<std::string>("getBuildStamp", [&] { return api::getBuildStamp(); });
}

double FFmpegKitExtended::createFFmpegSession(std::string command) noexcept {
  return invoke<double>("createFFmpegSession", [&] { return api::createFFmpegSession(command); });
}

double FFmpegKitExtended::createFFprobeSession(std::string command) noexcept {
  return invoke<double>("createFFprobeSession", [&] { return api::createFFprobeSession(command); });
}

double FFmpegKitExtended::createFFplaySession(std::string command) noexcept {
  return invoke<double>("createFFplaySession", [&] { return api::createFFplaySession(command); });
}

double FFmpegKitExtended::createMediaInformationSession(std::string command) noexcept {
  return invoke<double>("createMediaInformationSession", [&] { return api::createMediaInformationSession(command); });
}

void FFmpegKitExtended::executeSessionAsync(double sessionId, double timeoutMs) noexcept {
  invokeVoid("executeSessionAsync", [&] { api::executeSessionAsync(sessionId, timeoutMs); });
}

void FFmpegKitExtended::cancelSession(double sessionId) noexcept {
  invokeVoid("cancelSession", [&] { api::cancelSession(sessionId); });
}

std::string FFmpegKitExtended::getSessionJson(double sessionId) noexcept {
  return invoke<std::string>("getSessionJson", [&] { return api::getSessionJson(sessionId); });
}

void FFmpegKitExtended::releaseSessionHandle(double sessionId) noexcept {
  invokeVoid("releaseSessionHandle", [&] { api::releaseSessionHandle(sessionId); });
}

std::string FFmpegKitExtended::getSessionsJson(std::string kind) noexcept {
  return invoke<std::string>("getSessionsJson", [&] { return api::getSessionsJson(kind); });
}

std::string FFmpegKitExtended::getLastSessionJson(std::string kind) noexcept {
  return invoke<std::string>("getLastSessionJson", [&] { return api::getLastSessionJson(kind); });
}

std::string FFmpegKitExtended::getLogsJson(double sessionId, double fromIndex) noexcept {
  return invoke<std::string>("getLogsJson", [&] { return api::getLogsJson(sessionId, fromIndex); });
}

std::string FFmpegKitExtended::getStatisticsJson(double sessionId, double fromIndex) noexcept {
  return invoke<std::string>("getStatisticsJson", [&] { return api::getStatisticsJson(sessionId, fromIndex); });
}

std::string FFmpegKitExtended::getMediaInformationJson(double sessionId) noexcept {
  return invoke<std::string>("getMediaInformationJson", [&] { return api::getMediaInformationJson(sessionId); });
}

void FFmpegKitExtended::ffplayStart(double sessionId) noexcept {
  invokeVoid("ffplayStart", [&] { api::ffplayStart(sessionId); });
}

void FFmpegKitExtended::ffplayPause(double sessionId) noexcept {
  invokeVoid("ffplayPause", [&] { api::ffplayPause(sessionId); });
}

void FFmpegKitExtended::ffplayResume(double sessionId) noexcept {
  invokeVoid("ffplayResume", [&] { api::ffplayResume(sessionId); });
}

void FFmpegKitExtended::ffplayStop(double sessionId) noexcept {
  invokeVoid("ffplayStop", [&] { api::ffplayStop(sessionId); });
}

void FFmpegKitExtended::ffplaySeek(double sessionId, double seconds) noexcept {
  invokeVoid("ffplaySeek", [&] { api::ffplaySeek(sessionId, seconds); });
}

double FFmpegKitExtended::ffplayGetPosition(double sessionId) noexcept {
  return invoke<double>("ffplayGetPosition", [&] { return api::ffplayGetPosition(sessionId); });
}

void FFmpegKitExtended::ffplaySetPosition(double sessionId, double seconds) noexcept {
  invokeVoid("ffplaySetPosition", [&] { api::ffplaySetPosition(sessionId, seconds); });
}

double FFmpegKitExtended::ffplayGetDuration(double sessionId) noexcept {
  return invoke<double>("ffplayGetDuration", [&] { return api::ffplayGetDuration(sessionId); });
}

std::int32_t FFmpegKitExtended::ffplayGetVideoWidth(double sessionId) noexcept {
  return invoke<std::int32_t>("ffplayGetVideoWidth", [&] { return api::ffplayGetVideoWidth(sessionId); });
}

std::int32_t FFmpegKitExtended::ffplayGetVideoHeight(double sessionId) noexcept {
  return invoke<std::int32_t>("ffplayGetVideoHeight", [&] { return api::ffplayGetVideoHeight(sessionId); });
}

bool FFmpegKitExtended::ffplayIsPlaying(double sessionId) noexcept {
  return invoke<bool>("ffplayIsPlaying", [&] { return api::ffplayIsPlaying(sessionId); });
}

bool FFmpegKitExtended::ffplayIsPaused(double sessionId) noexcept {
  return invoke<bool>("ffplayIsPaused", [&] { return api::ffplayIsPaused(sessionId); });
}

void FFmpegKitExtended::ffplaySetVolume(double sessionId, double volume) noexcept {
  invokeVoid("ffplaySetVolume", [&] { api::ffplaySetVolume(sessionId, volume); });
}

double FFmpegKitExtended::ffplayGetVolume(double sessionId) noexcept {
  return invoke<double>("ffplayGetVolume", [&] { return api::ffplayGetVolume(sessionId); });
}

bool FFmpegKitExtended::ffplayHasVideoStream(std::string path) noexcept {
  return invoke<bool>("ffplayHasVideoStream", [&] { return api::ffplayHasVideoStream(path); });
}

void FFmpegKitExtended::enableRedirection() noexcept {
  invokeVoid("enableRedirection", [&] { api::enableRedirection(); });
}

void FFmpegKitExtended::disableRedirection() noexcept {
  invokeVoid("disableRedirection", [&] { api::disableRedirection(); });
}

void FFmpegKitExtended::setLogLevel(std::int32_t level) noexcept {
  invokeVoid("setLogLevel", [&] { api::setLogLevel(level); });
}

std::int32_t FFmpegKitExtended::getLogLevel() noexcept {
  return invoke<std::int32_t>("getLogLevel", [&] { return api::getLogLevel(); });
}

std::string FFmpegKitExtended::logLevelToString(std::int32_t level) noexcept {
  return invoke<std::string>("logLevelToString", [&] { return api::logLevelToString(level); });
}

void FFmpegKitExtended::setFontDirectory(std::string path, std::string mappingJson) noexcept {
  invokeVoid("setFontDirectory", [&] { api::setFontDirectory(path, mappingJson); });
}

void FFmpegKitExtended::setEnvironmentVariable(std::string name, std::string value) noexcept {
  invokeVoid("setEnvironmentVariable", [&] { api::setEnvironmentVariable(name, value); });
}

void FFmpegKitExtended::ignoreSignal(std::int32_t signal) noexcept {
  invokeVoid("ignoreSignal", [&] { api::ignoreSignal(signal); });
}

void FFmpegKitExtended::setAudioOutputDevice(std::string deviceName) noexcept {
  invokeVoid("setAudioOutputDevice", [&] { api::setAudioOutputDevice(deviceName); });
}

std::string FFmpegKitExtended::listAudioOutputDevices() noexcept {
  return invoke<std::string>("listAudioOutputDevices", [&] { return api::listAudioOutputDevices(); });
}

std::string FFmpegKitExtended::getFFmpegVersion() noexcept {
  return invoke<std::string>("getFFmpegVersion", [&] { return api::getFFmpegVersion(); });
}

std::string FFmpegKitExtended::getFFmpegArchitecture() noexcept {
  return invoke<std::string>("getFFmpegArchitecture", [&] { return api::getFFmpegArchitecture(); });
}

std::string FFmpegKitExtended::getVersion() noexcept {
  return invoke<std::string>("getVersion", [&] { return api::getVersion(); });
}

std::string FFmpegKitExtended::getPackageName() noexcept {
  return invoke<std::string>("getPackageName", [&] { return api::getPackageName(); });
}

std::string FFmpegKitExtended::getExternalLibraries() noexcept {
  return invoke<std::string>("getExternalLibraries", [&] { return api::getExternalLibraries(); });
}

std::string FFmpegKitExtended::getBundleType() noexcept {
  return invoke<std::string>("getBundleType", [&] { return api::getBundleType(); });
}

bool FFmpegKitExtended::isGpl() noexcept {
  return invoke<bool>("isGpl", [&] { return api::isGpl(); });
}

bool FFmpegKitExtended::isNonfree() noexcept {
  return invoke<bool>("isNonfree", [&] { return api::isNonfree(); });
}

std::string FFmpegKitExtended::getRegisteredCodecs() noexcept {
  return invoke<std::string>("getRegisteredCodecs", [&] { return api::getRegisteredCodecs(); });
}

std::string FFmpegKitExtended::getRegisteredEncoders() noexcept {
  return invoke<std::string>("getRegisteredEncoders", [&] { return api::getRegisteredEncoders(); });
}

std::string FFmpegKitExtended::getRegisteredDecoders() noexcept {
  return invoke<std::string>("getRegisteredDecoders", [&] { return api::getRegisteredDecoders(); });
}

std::string FFmpegKitExtended::getRegisteredMuxers() noexcept {
  return invoke<std::string>("getRegisteredMuxers", [&] { return api::getRegisteredMuxers(); });
}

std::string FFmpegKitExtended::getRegisteredDemuxers() noexcept {
  return invoke<std::string>("getRegisteredDemuxers", [&] { return api::getRegisteredDemuxers(); });
}

std::string FFmpegKitExtended::getRegisteredFilters() noexcept {
  return invoke<std::string>("getRegisteredFilters", [&] { return api::getRegisteredFilters(); });
}

std::string FFmpegKitExtended::getRegisteredProtocols() noexcept {
  return invoke<std::string>("getRegisteredProtocols", [&] { return api::getRegisteredProtocols(); });
}

std::string FFmpegKitExtended::getRegisteredBitstreamFilters() noexcept {
  return invoke<std::string>("getRegisteredBitstreamFilters", [&] { return api::getRegisteredBitstreamFilters(); });
}

std::string FFmpegKitExtended::getBuildConfiguration() noexcept {
  return invoke<std::string>("getBuildConfiguration", [&] { return api::getBuildConfiguration(); });
}

std::string FFmpegKitExtended::getBuildDate() noexcept {
  return invoke<std::string>("getBuildDate", [&] { return api::getBuildDate(); });
}

void FFmpegKitExtended::setSessionHistorySize(double size) noexcept {
  invokeVoid("setSessionHistorySize", [&] { api::setSessionHistorySize(size); });
}

double FFmpegKitExtended::getSessionHistorySize() noexcept {
  return invoke<double>("getSessionHistorySize", [&] { return api::getSessionHistorySize(); });
}

void FFmpegKitExtended::clearSessions() noexcept {
  invokeVoid("clearSessions", [&] { api::clearSessions(); });
}

std::string FFmpegKitExtended::registerNewFFmpegPipe() noexcept {
  return invoke<std::string>("registerNewFFmpegPipe", [&] { return api::registerNewFFmpegPipe(); });
}

void FFmpegKitExtended::closeFFmpegPipe(std::string path) noexcept {
  invokeVoid("closeFFmpegPipe", [&] { api::closeFFmpegPipe(path); });
}

double FFmpegKitExtended::messagesInTransmit(double sessionId) noexcept {
  return invoke<double>("messagesInTransmit", [&] { return api::messagesInTransmit(sessionId); });
}

void FFmpegKitExtended::enableDebugLog(double sessionId) noexcept {
  invokeVoid("enableDebugLog", [&] { api::enableDebugLog(sessionId); });
}

void FFmpegKitExtended::disableDebugLog(double sessionId) noexcept {
  invokeVoid("disableDebugLog", [&] { api::disableDebugLog(sessionId); });
}

bool FFmpegKitExtended::isDebugLogEnabled(double sessionId) noexcept {
  return invoke<bool>("isDebugLogEnabled", [&] { return api::isDebugLogEnabled(sessionId); });
}

std::string FFmpegKitExtended::getDebugLog(double sessionId) noexcept {
  return invoke<std::string>("getDebugLog", [&] { return api::getDebugLog(sessionId); });
}

void FFmpegKitExtended::clearDebugLog(double sessionId) noexcept {
  invokeVoid("clearDebugLog", [&] { api::clearDebugLog(sessionId); });
}

} // namespace winrt::FFmpegKitExtended
